### ISSUE TITLE
Pothos::Object support binary serialization

### ISSUE THREAD

**Issue by guruofquality** [pothosware/pothos-library#106]
Make use of binary serialization in Pothos::Object::serialize() and Pothos::Object::deserialize(). Binary serialization is faster.

Currently we use the portable text serialization, but there are many cases when binary is acceptable (for example same host, same architecture).

* This will require adding a parameter to de/serialize() to specify binary vs text. An API change

**Comment by guruofquality** [comment]
Follow up tasks:

* enableBinary() method for network source/sink blocks
* binary compatibility detection and enabling binary in network blocks in https://github.com/pothosware/pothos-library/blob/master/lib/Framework/TopologyNetworkFlows.cpp

**Comment by guruofquality** [comment]
I found a portable serialization from EOS that is an additional Boost archive. It claims to support binary serialization across architectures. I was able to test it between amd64 linux and arm32 linux. The code is now included as part of pothos-serialization. Modifications were needed to build it in properly.

* merge https://github.com/pothosware/pothos/commit/2865b4e380eeebfdba051110f5fe639e3e38a6fe
* eos portable archive: https://epa.codeplex.com/
